code/modules/mob/living/carbon/xenomorph/castes/baneling/baneling.dm

```diff
@@ -0,0 +1,95 @@
+/mob/living/carbon/xenomorph/baneling
+	caste_base_type = /mob/living/carbon/xenomorph/baneling
+	name = "Baneling"
+	desc = ""
+	icon = 'icons/Xeno/2x2_Xenos.dmi'
+	icon_state = "Baneling Walking"
+	bubble_icon = "alienleft"
+	health = 100
+	maxHealth = 100
+	plasma_stored = 50
+	tier = XENO_TIER_ONE
+	upgrade = XENO_UPGRADE_ZERO
+	pixel_x = -16
+	old_x = -16
+	/// reference to our pod
+	var/obj/structure/xeno/baneling_pod/pod_ref
+
+/// We do this to avoid a runtime with images assoc because baneling goes inside of their pod if they have one and there is no atom.z value inside of the pod
+/mob/living/carbon/xenomorph/baneling/on_death()
+	if(!isnull(pod_ref))
+		return
+	return ..()
+
+/// Delete the pod if we evolve or devolve
+/mob/living/carbon/xenomorph/baneling/finish_evolve()
+	if(!isnull(pod_ref))
+		pod_ref.baneling_ref = null
+		pod_ref.obj_destruction()
+		pod_ref = null
+	return ..()
+
+/mob/living/carbon/xenomorph/baneling/set_stat()
+	. = ..()
+	if(isnull(.))
+		return
+	if(. == CONSCIOUS && layer != initial(layer))
+		layer = MOB_LAYER
+
+/obj/structure/xeno/baneling_pod
+	name = "Baneling Pod"
+	desc = "A baneling pod, storing fresh banelings "
+	icon = 'icons/Xeno/2x2_Xenos.dmi'
+	icon_state = "Baneling Pod"
+	density = FALSE
```

Comment:
Why undense?

Reply:
Xenos HATE being bodyblocked by troll placements which I predict will become commonplace for a spammable T1 caste